TramsDataApi.Test/Integration/V4/IdentifiersV4IntegrationTests.cs

```diff
@@ -0,0 +1,197 @@
+
+namespace TramsDataApi.Test.Integration.V4;
+[Collection(ApiTestCollection.ApiTestCollectionName)]
+public class IdentifiersV4IntegrationTests
+{
+    private readonly HttpClient _client;
+    private readonly ApiTestFixture _apiFixture;
+
+    private readonly string _apiUrlPrefix = "/v4";
+
+    public IdentifiersV4IntegrationTests(ApiTestFixture fixture)
+    {
+        _apiFixture = fixture;
+        _client = fixture.Client;
+    }
+    
+    [Theory]
+    [InlineData(IdTypes.GroupID)]
+    [InlineData(IdTypes.UKPRN)]
+    [InlineData(IdTypes.GroupUID)]
+    public async Task Get_TrustIdentifiers_AndTrustExists_Returns_Ok(IdTypes idType)
+    {
+        using var context = _apiFixture.GetMstrContext();
+
+        var trustData = BuildSmallTrustSet();
+
+        context.Trusts.AddRange(trustData);
+        await context.SaveChangesAsync();
+
+        var selectedTrust = trustData.First();
+
+        var identifier = idType switch
+        {
+            IdTypes.GroupID => selectedTrust.GroupID,
+            IdTypes.UKPRN => selectedTrust.UKPRN,
+            IdTypes.GroupUID => selectedTrust.GroupUID,
+            _ => throw new ArgumentOutOfRangeException(nameof(idType), idType, null)
+        };
+
+        var trustResponse = await _client.GetAsync($"{_apiUrlPrefix}/identifier/{identifier}");
+        trustResponse.StatusCode.Should().Be(HttpStatusCode.OK);
+
+        var trustContent = await trustResponse.Content.ReadFromJsonAsync<TrustIdentifiers[]>();
+        trustContent.Length.Should().Be(1);
+        AssertIdentifierResponse(trustContent.First(), selectedTrust);
+    }
+    
+    [Theory]
+    [InlineData(IdTypes.GroupID)]
+    [InlineData(IdTypes.UKPRN)]
+    [InlineData(IdTypes.GroupUID)]
+    public async Task Get_TrustIdentifiers_AndDuplicateTrustsExists_Returns_Ok(IdTypes idType)
+    {
+        using var context = _apiFixture.GetMstrContext();
+
+        var trustData = BuildSmallDuplicateTrustSet(idType);
+
+        context.Trusts.AddRange(trustData);
+        await context.SaveChangesAsync();
+
+        var selectedTrust = trustData.First();
+        
+        var identifier = idType switch
+        {
+            IdTypes.GroupID => "GroupID",
+            IdTypes.UKPRN => "UKPRN",
+            IdTypes.GroupUID => "GroupUID",
+            _ => throw new ArgumentOutOfRangeException(nameof(idType), idType, null)
+        };
+
+        var trustResponse = await _client.GetAsync($"{_apiUrlPrefix}/identifier/{identifier}");
+        trustResponse.StatusCode.Should().Be(HttpStatusCode.OK);
+
+        var trustContent = await trustResponse.Content.ReadFromJsonAsync<TrustIdentifiers[]>();
+        trustContent.Length.Should().Be(3);
+        AssertIdentifierResponse(trustContent.First(), selectedTrust);
+    }
+    
+    [Theory]
+    [InlineData(IdTypes.GroupID)]
+    [InlineData(IdTypes.UKPRN)]
+    [InlineData(IdTypes.GroupUID)]
+    public async Task Get_TrustIdentifiers_AndNoOtherIdentifiersExist_Returns_Ok(IdTypes idType)
```

Comment:
Is this checking that it can handle null?

Reply:
Yes, just checking that other values can be null if they don't have values in the db